### PR TITLE
Added a ton of checks from the Dev Rob fork

### PR DESCRIPTION
**Pull request by 3pacgangstarap**
Spent a lot of time adding more locations to the tracker. Changes include:
added checks for scarecrow song
minor tweaks to location alias'
added treasure chest game keys
added empty pots and crates
added grass
added enemy drops
added wonderitems
added gossip stone drops
icons for pots and crates
also added new settings to the menu for all the new options

### CODE REVIEW

File: src/data/settings_presets/defaults.json
Comment (3pacgangstarap) [inline src/data/settings_presets/defaults.json:1]: you can probably ignore or undo these changes when merging, this was just my version of the defaults file that got uploaded

**Comment by mracsys** [comment]
Thanks a bunch for putting this together! I'll take a look over the weekend and likely merge next weekend after the standard tournament qualifiers are complete.

For timeline on the larger architectural changes, I'm targeting end of S7 brackets. Historically this has been around April/May each year. It'll address all open issues except #12 and maybe #17. I'm around 80% done, just keeping it under wraps until everything is done.

File: src/data/versions/dev6.0.41r-1.json
[the remark this answers is not in the document]
Comment (mracsys) [reply src/data/versions/dev6.0.41r-1.json:1]: Thanks! These will eventually be changed to use the same language as the upstream rando's UI. Setting, entrance, and location management got moved to https://github.com/mracsys/randomizer-graph-tool/tree/tracker-integration

File: src/data/settings_presets/defaults.json
Comment (mracsys) [reply src/data/settings_presets/defaults.json:1]: The defaults have previously been the highest shuffle settings possible in the RSL racing scene. As far as I know they are the heaviest users of the tracker. I'll make sure to update this before merging.

**Comment by 3pacgangstarap** [comment]
awesome. yeah this started as a personal project for use in my own runs cause I was testing out new mods like shuffle grass and enemy drops and nobody seemed to have all the checks for those marked down anywhere. both features are still in development, so more changes will probably need to be made as I'm doing testing. i already have a notepad filled with a few things i plan on tweaking. anyway i realized what i was doing would probably actually be pretty helpful to the community at large so i decided to fork your repository and here we are. One of the big changes i know already i want to make is reorder the lists to better integrate new checks into old areas. it's a little tedious having to scroll down to where all the grass and enemy drops are then back up to the other stuff in that part of the dungeon. anyway cheers

File: src/data/locations/death_mountain_crater.json [hunk 1]
```diff
@@ -1,726 +1,1215 @@
...
+          "required": false
+        },
+        {
+          "setting": "Tokensanity",
+          "value": "All",
+          "required": false
+        }
+      ]
+    },
+    "DMC Plant Bean Patch": {
```
Comment (mracsys) [inline src/data/locations/death_mountain_crater.json:324]: Can you remove the events from the location lists? It messes with the counter.

Also applies to the Plant Bean and Scarecrow events in other files.

I'd also be open to making these an option as long as they aren't included in the total location count.

File: src/data/versions/dev6.0.41r-1.json [hunk 2]
```diff
@@ -125,7 +125,8 @@
             ],
             "Shuffle Child Trade": [
                 "Off",
-                "On"
+                "Basic",
```
Comment (mracsys) [inline src/data/versions/dev6.0.41r-1.json:128]: If this is going to change from an on/off switch, it should be made a multi-select for every possible child trade item to align with the upstream randomizer setting.

File: src/data/versions/dev6.0.41r-1.json [hunk 3]
```diff
@@ -32,10 +32,10 @@
             ]            
         },
         "ER Settings": {
-            "open": true,
```
Comment (mracsys) [inline src/data/versions/dev6.0.41r-1.json:35]: I would prefer to keep ER options expanded by default as the default settings on page load is full ER. Less clicks to tone it down if not using a preset.

File: src/images/OoT_Jar_Icon.png
Comment (mracsys) [inline src/images/OoT_Jar_Icon.png:1]: Can you use this image instead?
pot

File: src/images/OoT_Crate_Icon.png
Comment (mracsys) [inline src/images/OoT_Crate_Icon.png:1]: Can you lighten this icon or replace with the attached?

crate

File: update.txt
Comment (mracsys) [inline update.txt:1]: Can you add these to a new Update History section of the readme?

File: src/data/locations/bottom_of_the_well.json
Comment (mracsys) [inline src/data/locations/bottom_of_the_well.json:1]: In the future it would be easier to review the data changes without the change in indents. Frankly these data files will disappear with v2 next year, so I'm not too hung up on it. If it looks good in testing I'll take it as-is.

**Review by mracsys — CHANGES_REQUESTED** [review]
Good work, some minor comments. I may have more after testing.

File: src/data/locations/bottom_of_the_well.json
Comment (3pacgangstarap) [reply src/data/locations/bottom_of_the_well.json:1]: i think the changes in indentation are an artifact of the JSON editor i was using. it probably messed with a lot of the formatting.

File: src/data/versions/dev6.0.41r-1.json (hunk 3 above)
Comment (3pacgangstarap) [reply src/data/versions/dev6.0.41r-1.json:35]: for sure, some tweaks i made mostly to my own personal preferences may have crept in, can be reverted easily enough

File: src/data/versions/dev6.0.41r-1.json (hunk 2 above)
Comment (3pacgangstarap) [reply src/data/versions/dev6.0.41r-1.json:128]: i agree, and same goes for shuffle adult trade. this was a really lazy implementation that assumes only you either have just the first item or all of them. not sure how to make it a multiselect but i'll look into it

File: src/images/OoT_Jar_Icon.png
Comment (3pacgangstarap) [reply src/images/OoT_Jar_Icon.png:1]: yeah, i had a hard time finding a decent image of the OoT pot. the one I eventually used I think came from a zelda wiki and was a render from OoT3D. this one looks good

File: src/images/OoT_Crate_Icon.png
Comment (3pacgangstarap) [reply src/images/OoT_Crate_Icon.png:1]: i think I like the 3d image better, but I can definitely lighten it up a bit. hows this?
OoT_Crate_Icon

File: src/data/locations/death_mountain_crater.json (hunk 1 above)
Comment (3pacgangstarap) [reply src/data/locations/death_mountain_crater.json:324]: is there a way to exclude them from the total count in addition to moving them under a new setting? haven't looked too deeply into the gui source code itself. i can separate them into a setting called 'events' or something similar. it would affect plant beans, scarecrow, and also the ones i added for the epona race and waking up talon in the master carpenter's house.

that last one I added because if there are no checks inside that interior it immediately gets hidden and can be hard to find where it is again on interior shuffle. it's also the only thing inside that interior that has any relevance to any play through.

if i made this 'events' setting i would probably also want to add the selling of masks to the guard, skull kid, etc as separate things. stuff under this umbrella becomes less about tracking item locations and more like reminders that this is something you might need to do while in this area.

**Comment by 3pacgangstarap** [comment]
i need to force myself to take a break from this for a little while, i went down a rabbit hole of being pretty obsessed over this for a few weeks and want to get back to other stuff. will come back to it during the new year to work on those requested changes

**Comment by mracsys** [comment]
No worries, take care of yourself! Happy holidays and hopefully see you around next year!

**Comment by 3pacgangstarap** [comment]
alright i think i addressed all the requested changes. took a little figuring out to implement multi checkboxes for child and adult trading settings. under the hood i made some edits to make it easier to make more settings into multi checkboxes in the future.

**Comment by mracsys** [comment]
Thanks again for all the work on this! Feel free to revert my commit in your branch to undo the ordering/naming changes made.

New version is live on my side.
